packages/react-components/src/shared/src/useChainedEventCallback.ts

```diff
@@ -2,16 +2,16 @@
 
 import { useEventCallback } from "./useEventCallback";
 
-export function createChainedFunction(...funcs) {
-    return (...args) => {
-        funcs
-            .filter(Boolean)
+export function createChainedFunction<T extends (...args: any[]) => void>(...funcs: (T | false)[]) {
+    return (...args: any[]) => {
+        (funcs
+            .filter(Boolean) as T[])
```

Comment:
This is weird, why the casting after filter? `filter` is a native function, shouldn't TS understand it returns an array?

Reply:
filter is a method that excludes some items from the list. it does not discriminate types for typescript sadly.